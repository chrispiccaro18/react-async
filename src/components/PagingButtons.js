import React from 'react';
import PropTypes from 'prop-types';

function PagingButtons({ handleNextButton }) {
  return (
    <button onClick={handleNextButton}>Next</button>
  );
}

PagingButtons.propTypes = {
  handleNextButton: PropTypes.func.isRequired
};

export default PagingButtons;
