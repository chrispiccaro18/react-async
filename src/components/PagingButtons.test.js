import React from 'react';
import { shallow } from 'enzyme';
import PagingButtons from './PagingButtons';

describe('PagingButtons comp tests', () => {
  it('creates a PagingButtons comp', () => {
    const handleNextButton = jest.fn(x => 42 + x);
    const handlePrevButton = jest.fn(x => 42 + x);
    const props = {
      handleNextButton,
      handlePrevButton,
      currentPage: 1,
      totalPages: 20
    };
    const wrapper = shallow(<PagingButtons {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
