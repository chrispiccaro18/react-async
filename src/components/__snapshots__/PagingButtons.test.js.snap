// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagingButtons comp tests creates a PagingButtons comp 1`] = `
<Fragment>
  <button
    disabled={true}
    onClick={[MockFunction]}
  >
    Previous
  </button>
  <span>
    1/20
  </span>
  <button
    disabled={false}
    onClick={[MockFunction]}
  >
    Next
  </button>
</Fragment>
`;
